demosplan/DemosPlanCoreBundle/Repository/UserRepository.php

```diff
@@ -533,7 +534,7 @@ public function getFirstUserInFhhnetByLogin(string $loginSuffix): User
     /**
      * @param User $user
      */
-    public function upgradePassword(UserInterface $user, string $newHashedPassword): void
+    public function upgradePassword(UserInterface|PasswordAuthenticatedUserInterface $user, string $newHashedPassword): void
```

Comment:
- what about using only PasswordAuthenticatedUserInterface. UserInterface is already extending from PasswordAuthenticatedUserInterface
- Maybe remove the doc comment?

Reply:
This is the `Symfony\Component\Security\Core\User\UserInterface`, not ours. The type hints are taken from the implemented method